### PR TITLE
Create project initial folders for 32_api_requests_in_cucumber_tests theme + README files

### PR DESCRIPTION
**Pull request by diogotia**
1) Introducing new directory andreidiogoti_32_api_requests_in_cucumber_tests
2) Readme files with descriptions " Cucumber Rest-Assured framework for API testing" with examples

### CODE REVIEW

**Comment by diogotia** [comment]
It is Draft PR - Work in progress.

**Comment by endalexei** [comment]
please remove toBeDeleted.file as it was temporary added to be able to create folders

**Comment by endalexei** [comment]
why do we need the _index file ?

File: ST/32_api_requests_in_cucumber_tests/docs/cucumber.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+and passes it as an argument to the {{% stepdef-body %}}.
+
+The number of parameters in the {{% stepdef-body %}} has to match the number of {{% expression-parameter %}}s in the expression. (If there is a mismatch, Cucumber will throw an error).
+
+## Data Tables
+
+Data tables from Gherkin can be accessed by using the DataTable object as the last parameter in a step definition.
+This conversion can be done either by Cucumber or manually.
+
+{{% text "java" %}}Depending on the table shape as one of the following collections:{{% /text %}}
```
Comment (endalexei) [inline ST/32_api_requests_in_cucumber_tests/docs/cucumber.md:23]: this {{% text "java" %}} appears on  page, is it intended ?
image

**Comment by diogotia** [comment]
> please remove toBeDeleted.file as it was temporary added to be able to create folders

toBeDeleted.file deleted

**Comment by diogotia** [comment]
> why do we need the _index file ? it useless here. Was deleted

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+title: API testing framework overview and setup
+subtitle: "Technology and frameworks"
+---
+
+# API Tests with Cucumber BDD Framework
+
+This is a project used for increase technical skills on API requests in Cucumber tests
```
Comment (tascaV) [inline ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md:8]: Remove line, not required

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+## Table of Contents
+
+1. [1. Project overview](#1-project-overview)
```
Comment (tascaV) [inline ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md:12]: remove redundant list:
[1.
[2.
[3.

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+2. [2. Technology and frameworks overview](#2-technology-and-frameworks-overview)
+3. [3. Framework setup](#3-framework-setup)
+
+## 1. Project overview
+
+1. Folder with project under test 'src/main/java/org/apitests'
```
Comment (tascaV) [inline ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md:18]: ~Folder with project under test~ -> Service under test

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+2. API testing framework 'src/test/java/org/apitests'
+3. Readme files with detailed descriptions 'src/docs'
+
+### 2. Technology and frameworks overview
+
+1. **Java Cucumber BDD framework**
```
Comment (tascaV) [inline ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md:24]: Cucumber

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    <details>
+   <summary>Scope</summary>
+   The framework uses Cucumber to implement BDD
+   We use the RequestSpecification class of rest assured for a one-time setup of the content headers, and other common
+   items used while making a new request.
+</details>
+
+2. **RestFull API**
```
Comment (tascaV) [inline ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md:32]: REST Assured

File: ST/32_api_requests_in_cucumber_tests/docs/framework_setup.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+   <properties>
+      <maven.compiler.source>21</maven.compiler.source>
+      <maven.compiler.target>21</maven.compiler.target>
+      <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+   </properties>
+   </poject>
+```
+
+## Setup RestAssured API Tests using the Cucumber BDD Style Test
```
Comment (tascaV) [inline ST/32_api_requests_in_cucumber_tests/docs/framework_setup.md:30]: L31-L33 can be removed

**Review by tascaV — CHANGES_REQUESTED** [review]
Please redo the documentation using snippets of code from example test framework. As well I will suggest to remove example test framework.
So to sum it up: embed source code within documentation itself.

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md (hunk 5 above)
Comment (diogotia) [reply ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md]: Updated

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+   <details>
+   <summary>Scope</summary>
+   The REST-Assured library provides a powerful and flexible API testing library that is used to make HTTP requests and verify the responses.
+   We use the RequestSpecification class of rest assured for a one-time setup of the content headers, and other common items used while making a new request.
+   </details>
+   
+3. **JUnit test runner**
```
[the remark this answers is not in the document]
Comment (diogotia) [reply ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md]: Updated

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md (hunk 6 above)
Comment (diogotia) [reply ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md]: updated on REST Assured

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md (hunk 4 above)
Comment (diogotia) [reply ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md]: updated

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md (hunk 3 above)
Comment (diogotia) [reply ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md]: updated

File: ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md (hunk 2 above)
Comment (diogotia) [reply ST/32_api_requests_in_cucumber_tests/docs/api-tests-with-cumber.md]: Removed

**Review by tascaV — COMMENTED** [review]
Looks good. Add some step defs examples and good to be merged

**Review by bodacd — APPROVED** [review]

**Comment by diogotia** [comment]
Step definition and Runner examples added

**Review by tascaV — APPROVED** [review]

**Review by rusakoviurii-Endava — APPROVED** [review]

**Review by olga-vozian — APPROVED** [review]

**Pull request merged**